pkg/storaged/client.js

```diff
@@ -206,6 +214,165 @@
     return false;
 }
 
+export async function btrfs_poll() {
+    const usage_regex = /used\s+(?<used>\d+)\s+path\s+(?<device>[\w/]+)/;
+    if (!client.uuids_btrfs_subvols)
+        client.uuids_btrfs_subvols = { };
+    if (!client.uuids_btrfs_usage)
+        client.uuids_btrfs_usage = { };
+    if (!client.btrfs_mounts)
+        client.btrfs_mounts = { };
+    if (!client.uuids_btrfs_default_subvol)
+        client.uuids_btrfs_default_subvol = { };
+    if (!client.uuids_btrfs_volume)
+        return;
+
+    if (!client.superuser.allowed) {
+        return;
+    }
+
+    const uuids_subvols = { };
+    const uuids_usage = { };
+    const btrfs_mounts = { };
+    const btrfs_default_subvol = { };
+    for (const uuid of Object.keys(client.uuids_btrfs_volume)) {
+        // In multi device setups MountPoints can be on either of the block devices, so try them all.
+        const blocks = client.uuids_btrfs_blocks[uuid];
+        const MountPoints = blocks.map(block => {
+            return client.blocks_fsys[block.path];
+        }).map(block_fsys => block_fsys.MountPoints).reduce((accum, current) => accum.concat(current));
+        const mp = MountPoints[0];
+        if (mp) {
+            let output;
+            const mount_point = utils.decode_filename(mp);
+            try {
+                output = await cockpit.spawn(["btrfs", "subvolume", "list", "-ap", mount_point], { superuser: true, err: "message" });
+            } catch (err) {
+                console.error(`unable to obtain subvolumes for mount point ${mount_point}`, err);
+                return;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20231211-210241-6861d8fd-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)